File: src/scitrader.marketdataserver/src/main/java/com/scitrader/marketdataserver/datastore/ITickAggregatorService.java
package com.scitrader.marketdataserver.datastore;

import com.scitrader.marketdataserver.common.Model.PriceBar;
import com.scitrader.marketdataserver.common.Model.PriceBarType;
import org.joda.time.DateTime;

import java.util.List;

public interface ITickAggregatorService {

  List<PriceBar> getPriceBars(String exchangeCode, String marketDataIdentifier, DateTime from, DateTime to, PriceBarType barType);
}

File: Kotlin/hello-kotlin-world/src/main/java/com/scitrader/marketdataserver/datastore/ITickAggregatorService.kt
package com.scitrader.marketdataserver.datastore

import com.scitrader.marketdataserver.common.Model.PriceBar
import com.scitrader.marketdataserver.common.Model.PriceBarType
import org.joda.time.DateTime

interface ITickAggregatorService {

    fun getPriceBars(
            exchangeCode: String,
            marketDataIdentifier: String,
            from: DateTime,
            to: DateTime,
            barType: PriceBarType,
            barTypeArg: Any): List<PriceBar>
}

